tiny_dnn/core/kernels/conv2d_op_nnpack.h

```diff
@@ -71,15 +67,24 @@ inline void conv2d_op_nnpack(const Tensor<float, S1> &in_data,
   float *output_ptr = out_data.host_pointer(0, 0);
 
   // TODO: embed it into a class
-  const size_t num_mkl_threads = 1;
+  const size_t num_mkl_threads = std::thread::hardware_concurrency();
   pthreadpool_t threadpool     = pthreadpool_create(num_mkl_threads);
```

Comment:
CNN_SINGLE_THREAD macro should be taken into account.
`threadpool` should be NULL in that case.